otx/hpo/utils.py

```diff
@@ -0,0 +1,124 @@
+"""HPO utility."""
+
+# Copyright (C) 2022 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+
+from typing import Any, List, Optional, Tuple, Union
+
+
+def check_type(
```

Comment:
This function is a common function also used by other codes. So, how about moving this under the `common`?

Reply:
I agree with you. But I don't know what exactly `common` means. Could you let me know which one is your meaning?

1. making otx/hpo/common directory and using it.
2. making otx/common directory and using it.
3. using otx/algorithm/common directory.

Reply:
Both of 2, 3 looks good to me. How about https://github.com/openvinotoolkit/training_extensions/blob/hpopt_consolidation/otx/algorithms/common/utils/utils.py ?

Reply:
Honestly, Topology looks a little wrong (otx/hpo uses otx/algorithm/common). But as you said, moving it to make it available to be used in other place is good choice. Maybe we can discuss about moving otx/algorithm/common to otx/common later. For now, I'll move it to otx/algorithm/common.